src/mrpro/operators/models/_TransientInversionRecovery.py

```diff
@@ -0,0 +1,190 @@
+        Mz(t) = M* + (M0 - M*)e^(-t * R1*)
+    (3) First inversion pulse (inversion_time_points[0])
+    (4) Time between inversion pulse and start of data acquisition, (time_inversion_adc), e.g. due to spoiler gradient
+        Mz(t) = M(1 - 2e^(-t * R1)) [M = M0 if no acquisitions before inversion pulse, otherwise it depends on (2)]
+    (5) Continuous data acquisition
+        Mz(t) = M* - (M + M*)e^(-t * R1*) [M = Mz(time_inversion_adc) using signal equation of (4)]
+    (6) Second inversion pulse (inversion_time_points[1]), followed by pause (time_inversion_adc) and next acquisition
+        block
+
+    (1)      (2)       (3)   (4)             (5)       (6)
+    |-----------------|INV|-------||------------------|INV|-------||----------------....
+
+    The model does not have to be evaluated for all the time points obtained during data acquisition. By defining the
+    start of the data acquisition, the time points of the inversion pulses and the gap between inversion pulses and the
+    first data acquisition, the sequence is fully described. This allows for the calculation of a signal from any
+    arbitrary subset of the data acquisition.
+
+
+    More information on this signal model can be found in:
+    Becker, K. M., Schulz-Menger, J., Schaeffter, T. & Kolbitsch, C.
+    Simultaneous high-resolution cardiac T1 mapping and cine imaging using model-based iterative image reconstruction.
+    Magn. Reson. Med. 81, 1080-1091 (2019). http://doi.wiley.com/10.1002/mrm.27474
+    """
+
+    def __init__(
+        self,
+        signal_time_points: torch.Tensor,
+        tr: float | torch.Tensor,
+        inversion_time_points: torch.Tensor,
+        time_inversion_adc: float | torch.Tensor,
+        first_adc_time_point: float | torch.Tensor | None = None,
+    ):
+        """Initialize continuous acquisition with inversion pulses.
+
+        Parameters
+        ----------
+        signal_time_points
+            time stamp of each acquisition
+        tr
+            repetition time
+        inversion_time_points
+            time stamp of each inversion
+        time_inversion_adc
+            time between inversion pulse and start of data acquisition
+        first_adc_time_point
+            time stamp of first acquisition
+        """
+        super().__init__()
+        tr = torch.Tensor(tr)
+        time_inversion_adc = torch.Tensor(time_inversion_adc)
+
+        self.signal_time_points = torch.nn.Parameter(signal_time_points, requires_grad=signal_time_points.requires_grad)
+        self.tr = torch.nn.Parameter(tr, requires_grad=tr.requires_grad)
+        self.time_inversion_adc = torch.nn.Parameter(time_inversion_adc, requires_grad=time_inversion_adc.requires_grad)
+
+        self.first_adc_time_point = torch.Tensor(first_adc_time_point) if first_adc_time_point is not None else None
+        self.index_before_first_inversion = torch.where(signal_time_points < inversion_time_points[0])[0]
```

Comment:
this should be register_buffer

Reply:
out of curiosity, what needs to be on the gpu? Do we need to be able to pu every variable in self on the gpu?